### DISCUSSION TITLE
How to validate entire form after resetting defalut value?

### DISCUSSION THREAD

**Discussion by nageshwari17** [react-hook-form/react-hook-form#9552] [Q&A]
Hi Team,

I have face one situation in my project that I need to validate the entire form after reseting default value . 

Please find CSB https://codesandbox.io/s/react-hook-form-reset-v7-r7m5s?file=/src/index.js 

In this first name is requred. I have resetting the form reset with value button . But in that first name is empty. 
As soon as I reset the form I need to validate the form . please help me with this .

Thanks in Advance.

**Comment by Moshyfawn** [comment]
You can simply call [the `trigger` form method](https://react-hook-form.com/api/useform/trigger) after [the `reset`](https://react-hook-form.com/api/useform/reset)
```diff
const { trigger } = useForm()

<button
  onClick={() => {
    reset(data)
+   trigger();
  }}
>
  Reset
</button>
```

**Reply by nageshwari17** [reply]
Hi @Moshyfawn ,

I have updated my CSB [https://codesandbox.io/s/react-hook-form-reset-v7-forked-tfulgv?file=/src/index.js](https://codesandbox.io/s/react-hook-form-reset-v7-forked-tfulgv?file=/src/index.js)

Its not working as ecpected . Please help me

**Reply by nageshwari17** [reply]
Hi guys, 

Is any input on this ?

**Comment by Moshyfawn** [comment]
You can do
```jsx
<form onReset={() => trigger()}>...</form>
```

**Reply by nageshwari17** [reply]
@Moshyfawn this is also not helping if value is pressent 

please check here https://codesandbox.io/s/react-hook-form-reset-v7-forked-vq8jcb?file=/src/index.js.

Is any callback function that I can trigger after reset ?

**Reply by nageshwari17** [reply]
Hi team , Please can any one help with this .
